### PR TITLE
Revised TextInput.md

### PR DESCRIPTION
**Pull request by kpoulincoveo**
### Proposed Changes

Revised all texts to match current state of the text inputs.
It's a first draft and more guidelines will be added later on if needed.

The best practices appears in quotes format as an attempt to style them within a grey box, not sure if how it would like in the real website though...

The description of the component (which should appear on the tile and in the header of its page) is commented in this file so that you guys have access.

### Potential Breaking Changes

I wasn't able to dismiss the hyperlink on the email, so please help. 

### Acceptance Criteria

Please make sure that the information here is compliant with the current behavior. If not, indicate how. 
If there is no specific code that prevents a user to apply such guidelines, we can keep it there.

This I don't know ⬇️ 
-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/website/docs/TextInput.md [hunk 1]
```diff
@@ -1,20 +1,43 @@
-## Guidelines
+[Description]: # 
+[Text inputs allow to enter and edit short texts, such as names, emails or passwords.]: # 
+>
+> ### Best Practices :bulb:
```
Comment (aiheon) [inline packages/website/docs/TextInput.md:4]: I think we'll need to add svg or icons in the repo if we want to use them in the guidelines :thinking: 
:bulb: is not rendering in the demo.

Comment (kpoulincoveo) [reply packages/website/docs/TextInput.md:4]: Don't bother, I will remove them, it was an hopeful test but not required for now.

Comment (gdostie) [reply packages/website/docs/TextInput.md]: real emojis should work

File: packages/website/docs/TextInput.md [hunk 2]
```diff
@@ -1,20 +1,48 @@
...
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases. 
+The minimum width is 320px and can be incremented by multiples of 8.
 
-### Behaviour
+---
+### Labeling
+Keep labels short, under three words.
```
Comment (kpoulincoveo) [inline packages/website/docs/TextInput.md:12]: Suggestion:  Keep labels short, **under three words**.

File: packages/website/docs/TextInput.md [hunk 3]
```diff
@@ -1,20 +1,48 @@
...
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases. 
+The minimum width is 320px and can be incremented by multiples of 8.
 
-### Behaviour
+---
+### Labeling
+Keep labels short, under three words. 
+
+Use descriptive labels. Avoid action verbs. (E.g. Use "Email" instead of "Write your email")
```
Comment (kpoulincoveo) [inline packages/website/docs/TextInput.md:14]: Suggestions: 
Use descriptive labels without action verbs. 
- E.g. Use "Email" ...

File: packages/website/docs/TextInput.md [hunk 4]
```diff
@@ -1,20 +1,48 @@
...
+
+Help text should be **short one-liner**. 
+There are two ways to provide help text guidance: 
+ 
+- Written instructions : "Provide your business email." 
+- Example given: "E.g.: johndoe@acme.com" 
+
+A combination of both is also acceptable. 
+
+Text input should be self-explanatory as much as possible.
```
Comment (kpoulincoveo) [inline packages/website/docs/TextInput.md:28]: Suggestion: 
The label of the text input should be self-explanatory as much as possible.

File: packages/website/docs/TextInput.md [hunk 5]
```diff
@@ -1,20 +1,48 @@
...
--   Display an inline validation message when the user is done interacting with the input.
+If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.
 
 ---
+### Related Components
+If your use case doesn't match these guidelines, consider using alternate component instead. 
+ 
+- [Text Area](#/components/TextArea) when you require more than one line of characters.
+- [Numeric Input](#/components/NumericInput) when you restrict characters to only numerical values.
+- [Single Select](#/input/SingleSelect) when you instead want user to select an option from a predefined list.
```
Comment (kpoulincoveo) [inline packages/website/docs/TextInput.md:45]: Correction:
#/input/TextArea
#/input/NumericInput

File: packages/website/docs/TextInput.md [hunk 6]
```diff
@@ -1,20 +1,49 @@
...
 
-### Restrictions & Size
+## Best Practices
+Use text inputs when the user must enter one **short line of text.**
 
--   Users should be informed of any input restrictions, such as restrictions on character types or string length.
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases.
```
Comment (sallainCoveo) [inline packages/website/docs/TextInput.md:7]: [Nitpick] if it's best practice I don't think you have to specify that this guideline apply in most case. 
```suggestion
The width of the text input should be proportional to the expected user input and large enough to display the entire text. 
```

**Review by gdostie — APPROVED** [review]
Added the description in the right place

**Review by sallainCoveo — APPROVED** [review]

File: packages/website/docs/TextInput.md [hunk 7]
```diff
@@ -1,20 +1,47 @@
...
 
-### Restrictions & Size
+Use text inputs when the user must enter one **short line of text.**
 
--   Users should be informed of any input restrictions, such as restrictions on character types or string length.
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text, in most cases.
+The minimum width is 320px and can be incremented by multiples of 8.
```
[the remark this answers is not in the document]
Comment (kpoulincoveo) [reply packages/website/docs/TextInput.md:6]: I will remove it for now. thanks

File: packages/website/docs/TextInput.md [hunk 8]
```diff
@@ -1,20 +1,46 @@
...
+The label of the text input should be self-explanatory as much as possible.
+Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.
+
+## Feedback and validation
+
+If some characters are restricted, consider indicating it in an help text instead of relying only on the validation message to inform the user about them.
+
+If the provided information isn't compliant with the requirements, make sure that the validation feedback clearly indicates how to fix it in the text of the validation message.
+
+If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.
```
Comment (jlevesquecoveo) [inline packages/website/docs/TextInput.md:36]: @kpoulincoveo It's unclear to me what we mean here.
"If the information is optional" = if it's optional that the user provides text, right?
"the text input will automatically be tagged" = the placeholder contains `(optional)`? 🤔

File: packages/website/docs/TextInput.md [hunk 9]
```diff
@@ -1,20 +1,46 @@
...
 
 ---
 
 ## Related Components
 
-[Text Area](#/components/TextArea)
+If your use case doesn't match these guidelines, consider using alternate component instead.
```
Comment (jlevesquecoveo) [inline packages/website/docs/TextInput.md:42]:
```suggestion
If your use case doesn't match the guidelines above, consider using one of the following components instead:
```

File: packages/website/docs/TextInput.md [hunk 10]
```diff
@@ -1,20 +1,46 @@
...
 
-[Numeric Input](#/components/NumericInput)
+-   [Text Area](#/form/TextArea) when you require more than one line of characters.
```
Comment (jlevesquecoveo) [inline packages/website/docs/TextInput.md:44]: @kpoulincoveo Again, just to confirm my understanding: does this mean that the user MUST provide 2+ lines, or that the information that they input MAY stand on 2+ lines (ex: two mapping rules, each on one line).

File: packages/website/docs/TextInput.md [hunk 11]
```diff
@@ -1,20 +1,46 @@
...
+-   [Numeric Input](#/form/NumericInput) when you restrict characters to only numerical values.
+-   [Single-Select](#/form/SingleSelect) when you instead want user to select an option from a predefined list.
```
Comment (jlevesquecoveo) [inline packages/website/docs/TextInput.md:46]:
```suggestion
-   [Single select](#/form/SingleSelect) to have users select an option from a list instead.
```

**Review by jlevesquecoveo — CHANGES_REQUESTED** [review]

File: packages/website/docs/TextInput.md [hunk 12]
```diff
@@ -1,20 +1,47 @@
...
 
--   Users should be informed of any input restrictions, such as restrictions on character types or string length.
--   The text input area width should suggest the expected size of the user input. Try to forsee the text input length so as to avoid cutting it off. The text input area should be large enough to display the whole text input in most cases.
--   Avoid setting a character limit, or make sure to set it high enough to avoid constraining most user entries. If this is not possible, let users know how many characters are left.
--   Ensure that users with small screens can still enter all required information.
+The width of the text input should be proportional to the expected user input and large enough to display the entire text.
+The minimum width is 320px and can be incremented by multiples of 8.
 
-### Behaviour
+## Labeling
```
[the remark this answers is not in the document]
Comment (kpoulincoveo) [reply packages/website/docs/TextInput.md:7]: As I said in Checkbox. Text is too general. We'll keep labeling for now.

File: packages/website/docs/TextInput.md [hunk 13]
```diff
@@ -1,20 +1,47 @@
...
+Help text should be **short one-liner**.
+There are two ways to provide help text guidance:
+
+-   Written instructions : "Provide your business email."
+-   Example given : "E.g.: johndoe@acme.com"
+
+A combination of both is also acceptable.
+
+The label of the text input should be self-explanatory as much as possible.
+Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.
```
[the remark this answers is not in the document]
Comment (kpoulincoveo) [reply packages/website/docs/TextInput.md:28]: What I meant here is that title and description should only be added when additional guidance is critical to the user experience. 

Revision changes:
Add a title and description when providing additional guidance is critical for the user to understand his task.

File: packages/website/docs/TextInput.md (hunk 10 above)
Comment (kpoulincoveo) [reply packages/website/docs/TextInput.md:44]: It's more like a paragraph than two distinctive lines.

File: packages/website/docs/TextInput.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (jlevesquecoveo) [reply packages/website/docs/TextInput.md:26]: Ah, I see, thanks!
```suggestion
Add a title and description when providing additional guidance is critical for the user to understand why the information is required or how the information will be used.
```

File: packages/website/docs/TextInput.md [hunk 14]
```diff
@@ -1,20 +1,44 @@
...
+The label of the text input should be self-explanatory.
+Add a title and description when providing additional guidance is critical to the user to understand why the information is required or how the information will be used.
+
+## Feedback and Validation
+
+If there are character restrictons, consider indicating them in the help text. Avoid relying only on the validation message to inform the user about them.
+
+If the user provides text that doesn't meet the requirements, the validation message should clearly indicate how to fix it.
+
+If the information is optional, the text input will be automatically tagged as so to indicate that it can be left blank.
```
Comment (jlevesquecoveo) [inline packages/website/docs/TextInput.md:34]:
```suggestion
If providing the information is optional, the text input will be automatically tagged as such, indicating that the input can be left blank.
```

File: packages/website/docs/TextInput.md (hunk 10 above)
Comment (jlevesquecoveo) [reply packages/website/docs/TextInput.md:42]: OK so how about:
```suggestion
-   [Text Area](#/form/TextArea) to have users enter longer text.
```

**Review by jlevesquecoveo — APPROVED** [review]

**Pull request merged**